### ISSUE TITLE
how did I use mother wavelet set knots 

### ISSUE THREAD

**Issue by Iveshey** [scatseisnet/scatseisnet#29]
Hi developers
I have a question want to ask you

I follow your release of open source, and you mention you initialized a Gabor wavelet that learns the parameters governing the shape of the wavelets with respect to the clustering loss. How did you accomplish it? Because you just set the octaves, resolution, and quality. You did not say something about how you let Mother Wavelet learn to apply the convolution.

Thank you

**Comment by leonard-seydoux** [comment]
Hi @Iveshey,

The current repository does not feature a learnable wavelet transformation. This functionality was available in the [scanet](https://github.com/leonard-seydoux/scatnet) repository, which was developed using TensorFlow 1. However, it was discontinued following the release of TensorFlow 2. If your hardware is compatible, you can still download and install the package to run the original code. Unfortunately, we are unable to provide support for this version of the package.

If you are interested in updating the previous repository to TensorFlow 2, it is technically possible, but be aware that we have identified several significant challenges without clear solutions. As a result, we do not plan to continue developing the learnable wavelet package and will instead focus on the existing non-learnable approach, utilizing pre-defined wavelets. From a scientific perspective, the Gabor wavelet still allows for the achievement of high-quality results.

We have decided to mark this issue as "won't fix". However, we welcome further discussion on the topic if you're interested.

Best,
Leonard

**Comment by Iveshey** [comment]
Hi developers,
Thank you for your response.
From a scientific perspective, the Gabor wavelet still allows for high-quality results. I want to ask you about your mention of the high-quality results in the Gabor wavelet. In addition to the Gabor wavelet, did you have to try another wavelet like Haar or db4 wavelet? Your seismic data is for a single case in the paper you published, and you released the example on GitHub. Still, our seismic data is about random rockfall cases. So, whether you have to train another wavelet, if you do not, how did you get the parameter(octaves, resolution, and quality)  optimization?

Best,
Iveshey

**Comment by leonard-seydoux** [comment]
Hi @Iveshey,

So far, we have not tried another wavelet since we consider that many parts in the processing chain of waveform clustering can dominate the results over wavelet design (dimension reduction, window size, clustering algorithm). Note that we have now published several papers using this algorithm on various datasets; please check the reference section of https://scatseisnet.readthedocs.io/en/latest/ for more information. All studies employ a complex Morlet wavelet.

About the design of the architecture, we do not optimize it. We declare the filter banks based on a priori (physics-based) information we have. More about it in the paper: Steinmann et al. (2022). Hierarchical exploration of continuous seismograms with unsupervised learning. Journal of Geophysical Research: Solid Earth, 127(1), e2021JB022455. https://doi.org/10.1029/2021JB022455.

Note that the main goal of switching from learnable to fixed-shape wavelets is also to avoid having to optimize a problem that does not find a simple solution already.

Best,
Leonard
